### PR TITLE
[dynamic_layouts] Added an example for wrap layout

### PR DESCRIPTION
**Pull request by snat-s**
Added an example for the wrap layout:

https://user-images.githubusercontent.com/65375294/188182176-86a9c1a7-bd32-484b-8e50-4b82f501d42d.mov

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like some checks are unhappy.

File: packages/dynamic_layouts/example/lib/wrap_layout_example.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/// The wrap example
+class WrapExample extends StatefulWidget {
+  /// The constructor for the wrap example
+  const WrapExample({super.key});
+
+  @override
+  State<WrapExample> createState() => _WrapExampleState();
+}
+
+class _WrapExampleState extends State<WrapExample> {
```
Comment (goderbauer) [inline packages/dynamic_layouts/example/lib/wrap_layout_example.dart:16]: Could be a stateless widget?

File: packages/dynamic_layouts/example/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+void main() {
+  runApp(const MyApp());
+}
+
+/// Main example
+class MyApp extends StatelessWidget {
+  /// Main example constructor.
+  const MyApp({super.key});
+
+  // This widget is the root of your application.
```
Comment (goderbauer) [inline packages/dynamic_layouts/example/lib/main.dart:17]: this comment can be removed.

File: packages/dynamic_layouts/example/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+name: example
+description: A new Flutter project.
+
+# The following line prevents the package from being accidentally published to
```
Comment (goderbauer) [inline packages/dynamic_layouts/example/pubspec.yaml:4]: you can basically remove all the comments in this file.

File: packages/dynamic_layouts/example/lib/wrap_layout_example.dart [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:dynamic_layouts/dynamic_layouts.dart';
```
Comment (goderbauer) [inline packages/dynamic_layouts/example/lib/wrap_layout_example.dart:4]: nit: add blank line after license header.

File: packages/dynamic_layouts/example/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:flutter/material.dart';
```
Comment (goderbauer) [inline packages/dynamic_layouts/example/lib/main.dart:4]: nit: add blank line after license header.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
can you resolve the merge conflicts?

**Comment by goderbauer** [comment]
This looks good, but one of the newly added tests appears to be failing. Any chance you can fix that up? Thanks!

**Comment by snat-s** [comment]
Hi! For sure

**Review by Piinks — APPROVED** [review]
This LGTM once the test failure in `widget_test.dart` is fixed. You can probably just delete that file, since the code it was testing has been removed.

**Pull request merged**